woocommerce/google-listings-and-ads
# Connect products report page to the data source of report API

### Changes proposed in this Pull Request:

Closes #244 . This PR relies on the `feature/242-244-chart-section` branch of #568 

- **CompareProductsTableCard**
    - align to data schema of report API
    - handle loading and empty status
    - some props adjustments
    - apply store's currency setting to format values
- **SummarySection**
    - align to data schema of report API
    - handle loading status
    - Move to `.~/reports` folder as a shared component
- Connect products report page to the data source of report API
- Remove mocked products report data

### Screenshots:

![Kapture 2021-05-11 at 11 01 48](https://user-images.githubusercontent.com/17420811/117751994-9c069a80-b248-11eb-939f-449ab9d24333.gif)

### Detailed test instructions:

💡. For easier mock testing, I pushed [another branch with some mock code](https://github.com/woocommerce/google-listings-and-ads/compare/feature/244-connect-data...feature/244-products-report-mock-local), which located at:

- Mock paid data
    https://github.com/woocommerce/google-listings-and-ads/blob/aaea8311f05fd624969ed42a871e82eb4638cf64/src/API/Site/Controllers/Ads/ReportsController.php#L84-L230
- Mock free data
    https://github.com/woocommerce/google-listings-and-ads/blob/aaea8311f05fd624969ed42a871e82eb4638cf64/src/API/Site/Controllers/MerchantCenter/ReportsController.php#L83-L197

🩺 . Test with the UI

1. Checkout `244-products-report-mock-local` branch if want to test with local mocked data
2. Head to the Products Report page `/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Freports%2Fproducts`
3. Use functions on this page, e.g. filers, sorting, it should fetch data from report API if the report query hasn't loaded before
4. All data displayed on this page should be sourced from report API
